recipe/common.php

```diff
@@ -46,41 +47,142 @@
  * Configuration
  */
 
-set('keep_releases', 5);
+set('keep_releases', 5, function ($value) {
+    return is_integer($value) && $value >= -1;
+});
 
-set('repository', ''); // Repository to deploy.
+set('repository', '', function ($value) {
+    return is_string($value); 
+}); // Repository to deploy.
 
-set('shared_dirs', []);
-set('shared_files', []);
+set('shared_dirs', [], function ($value) {
+    if (!is_array($value)) {
+        return false;
+    }
+    foreach ($value as $dir) {
+        if (!is_dir($dir)) {
```

Comment:
I think this is not generally true. Some may want to add unexisting dirs.

Reply:
Thanks! I am removing this `foreach`.